src/bench.c

```diff
@@ -623,10 +617,8 @@ void gather_results(struct bench_results *results)
 		MPI_SUM, 0, MPI_COMM_WORLD);
 	MPI_Reduce(&results->virtual, &combined.virtual, 1, MPI_LONG,
 		MPI_SUM, 0, MPI_COMM_WORLD);
-	MPI_Reduce(&results->crypts.lo, &combined.crypts.lo, 1, MPI_UNSIGNED,
-		MPI_SUM, 0, MPI_COMM_WORLD);
-	MPI_Reduce(&results->crypts.hi, &combined.crypts.hi, 1, MPI_UNSIGNED,
-		MPI_SUM, 0, MPI_COMM_WORLD);
+	MPI_Reduce(&results->crypts, &combined.crypts, 1,
+	           MPI_UNSIGNED_LONG_LONG, MPI_SUM, 0, MPI_COMM_WORLD);
```

Comment:
Really old/odd MPI environments may not have MPI_UNSIGNED_LONG_LONG, it's optional. We'll see if that's a problem in real life. If it is, we'll just add it to autoconf and disable MPI.

Reply:
Why don't we simply keep the two `MPI_Reduce` lines using `& 0xffffffffU` and `>> 32`, respectively? Sounds like a cheap enough price for not relying on optional functionality of the MPI library.

Reply:
Unless I'm confused it will (or may) produce incorrect result if we eg. have only two nodes, and their figures are just above and just below 0x100000000? We've had that problem until now though.

Reply:
We could do so only when MPI_UNSIGNED_LONG_LONG isn't supported though. Let's see if this is a real problem or not.

Reply:
OK. I didn't realize this was fixing a maybe-real problem.

Reply:
I opted to fix this in 9e02be954 before anyone complained: For systems lacking 64-bit reduce we fall back to old buggy behavior of adding halves without carry. Is there even any way to work around that problem? I mean, we have *n* nodes, potentially thousands.

Anyway it only affects --test under MPI so it's cosmetical.